test/unit/specs/utils/promiseAllObject.spec.js

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import promiseAllObject from "../../../../src/utils/promiseAllObject";
+
+const createFoodLibrary = () => {
+  return {
+    fruits: [
+      {
+        name: "Apple",
+        calories: 95
+      },
+      {
+        name: "Avocado",
+        calories: Promise.resolve(234)
+      },
+      {
+        name: "Banana",
+        calories: 133
+      },
+      Promise.resolve("Cherry")
+    ],
+    vegetables: Promise.resolve([
```

Comment:
What happens if the promise is rejected? Can you add a test for that scenario?
Can you add a test for a promise that isn't resolved instantly? Maybe that doesn't matter... idk.

Reply:
The promise returned from `promiseAllObject` should be rejected. It's a good idea to add a test for that though. I will do that.

Reply:
Regarding this:

> Can you add a test for a promise that isn't resolved instantly?

I think it helps to understand that promises are ALWAYS asynchronous, regardless of whether they're resolved immediately or not. In other words, even though we resolve the promise here immediately, any callbacks passed through a `.then()` on the promise will always be called _asynchronously_. So it doesn't really matter whether it takes .001 milliseconds for the callback to be called or 1000 seconds, it's guaranteed to be asynchronous, which is the important part, and we can reasonably assume that the difference in the amount of time won't cause a difference in behavior within `promiseAllObject`.